code/modules/awaymissions/mission_code/dsbssigma.dm

```diff
@@ -0,0 +1,735 @@
+
+/obj/effect/mob_spawn/human/corpse/away/dsbssigma/operative/miner
+	shoes = /obj/item/clothing/shoes/workboots/mining
+	head = /obj/item/clothing/head/hardhat
+	suit = /obj/item/clothing/suit/storage/hazardvest
+	belt = /obj/item/storage/belt/military/traitor
+
+/obj/effect/mob_spawn/human/corpse/away/dsbssigma/monkey
+	mob_species = /datum/species/monkey
+
+/obj/effect/mob_spawn/human/corpse/away/dsbssigma/monkey/equip(mob/M)
+	..()
+	M.reagents.add_reagent("formaldehyde", 1)
+
+/mob/living/simple_animal/hostile/carp/dumpstershark
+	icon = 'icons/mob/alienqueen.dmi'
+	name = "Dumpster Shark"
+	desc = "A ferocious, fang bearing creature that resembles a shark. The bane of space-trash-men everywhere."
+	icon_state = "megacarp"
+	icon_living = "megacarp"
+	icon_dead = "megacarp_dead"
+	icon_gib = "megacarp_gib"
+	gold_core_spawnable = NO_SPAWN
+	maxHealth = 120
+	health = 120
+	pixel_x = -16
+	mob_size = MOB_SIZE_LARGE
+	wander = FALSE
+	random_color = FALSE
+	butcher_results = list(/obj/item/toddler = 1, /obj/item/reagent_containers/food/snacks/carpmeat = 2)
+	obj_damage = 80
+	melee_damage_lower = 30
+	melee_damage_upper = 20
+	move_to_delay = 5
+	robust_searching = TRUE
+
+/mob/living/simple_animal/hostile/carp/stationary
+	wander = FALSE
+
+/mob/living/simple_animal/hostile/syndicate/ranged/dsbssigma
+	name = "Staff Sgt Hope"
+	desc = "Battered, wounded and on the verge of insanity, but determined to survive. As the saying goes, Hope dies last!"
+	robust_searching = TRUE
+	stat_attack = UNCONSCIOUS
+	loot = list()
+	maxHealth = 200
+	health = 200
+	del_on_death = TRUE
+	ranged = TRUE
+	rapid = 1
+	retreat_distance = 3
+	minimum_distance = 1
+	melee_damage_lower = 20
+	melee_damage_upper = 25
+	icon_state = "syndicate_pistol_alt"
+	icon_living = "syndicate_pistol_alt"
+	projectiletype = /obj/item/projectile/bullet/armourpiercing
+	casingtype = /obj/item/ammo_casing/c10mm/ap
+	projectilesound = 'sound/weapons/gunshots/gunshot_pistol.ogg'
+	environment_smash = FALSE
+	see_in_dark = 8
+
+/mob/living/simple_animal/hostile/syndicate/ranged/dsbssigma/death()
+	new /obj/effect/gibspawner/human(get_turf(src))
+	return ..()
+
+/mob/living/simple_animal/hostile/alien/stationary
+	wander = FALSE
+
+/mob/living/simple_animal/hostile/alien/drone/stationary
+	wander = FALSE
+
+/mob/living/simple_animal/hostile/alien/sentinel/stationary
+	wander = FALSE
+
+/mob/living/simple_animal/hostile/alien/queen/large/dsbssigma
+	wander = FALSE
+	var/hasdied = FALSE
+
+/mob/living/simple_animal/hostile/alien/queen/large/dsbssigma/death()
+	if(can_die() && !hasdied)
+		hasdied = TRUE
+		UnlockBlastDoors("dsbss_overseersquarters_blast")
+		for(var/mob/M in GLOB.player_list)
+			if(M.z == z)
+				to_chat(M, "<span class='notice'>You hear a distant sound of a blast door opening.</span>")
+				M.playsound_local(null, 'sound/machines/blastdoor.ogg', 10, 1)
+	return ..()
+
+/mob/living/simple_animal/hostile/alien/queen/large/dsbssigma/proc/UnlockBlastDoors(target_id_tag)
```

Comment:
This is duplicate code with `/mob/living/simple_animal/hostile/poison/terror_spider/proc/UnlockBlastDoors`